nano/node/ipc/ipc_server.cpp

```diff
@@ -568,26 +554,6 @@ std::optional<std::uint16_t> socket_transport<ACCEPTOR_TYPE, SOCKET_TYPE, ENDPOI
 
 }
 
-/**
- * Awaits SIGHUP via signal_set instead of std::signal, as this allows the handler to escape the
- * Posix signal handler restrictions
- */
-void await_hup_signal (std::shared_ptr<boost::asio::signal_set> const & signals, nano::ipc::ipc_server & server_a)
```

Comment:
Is this removing the signal handler to reload the node config?

I don't think this ever worked correctly anyway.

Reply:
Yes, it's removing it. It was hijacking node.io_context to run (and seemed to be breaking its lifetime, preventing clean shutdown) and there was no immediately obvious place to put it. There is already a dedicated signal manager class that wraps asio stuff and provides better interface and safety. I didn't have enough time to investigate and fix it, that's why I decided to abandon it. Similar story with the `ipc_server::stop` workaround - ipc is doing things differently than the rest of the node for no reason, which makes it impossible to quickly fix.